src/libs/actions/App.js

```diff
@@ -52,10 +53,24 @@ function setCurrentURL(url) {
 * @param {String} locale
 */
 function setLocale(locale) {
-    if (currentUserAccountID) {
-        DeprecatedAPI.PreferredLocale_Update({name: 'preferredLocale', value: locale});
+    // If user is not signed in, change just locally.
+    if (!currentUserAccountID) {
+        Onyx.merge(ONYXKEYS.NVP_PREFERRED_LOCALE, locale);
+        return;
     }
-    Onyx.merge(ONYXKEYS.NVP_PREFERRED_LOCALE, locale);
+
+    // Optimistically change preferred locale
+    const optimisticData = [
+        {
+            onyxMethod: 'merge',
+            key: ONYXKEYS.NVP_PREFERRED_LOCALE,
+            value: locale,
+        },
+    ];
+
+    API.write('UpdatePreferredLocale', {
+        value: locale,
```

Comment:
NAB, kind of odd that there is an API parameter called `value` ? 😄 Curious whether it should be more specific but definitely would not block on this comment.